### PR TITLE
Add value-description table to `process` key

### PR DESCRIPTION
**Pull request by generalmimon**
As [discussed on Gitter](https://gitter.im/kaitai_struct/Lobby?at=5ece6f922280c80cbfd03795) with @GreyCat 

The fastest way to view it is to go on http://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/kaitai-io/ksy_schema/process-table/ksy_schema.json and find the table in the `seq/process` key description.

I'm a little disappointed with the way how Markdown handles formatting in table cells. AFAIK you can't just start a new paragraph while staying in the table cell as you normally would (with two newlines: `\n\n`), so I had to fall back to the ugly `<p></p>` thing. But as a result, it looks pretty good, and that's what matters. I'm quite satisfied with it 🙂

### CODE REVIEW

**Comment by generalmimon** [comment]
@dgelessus, can I also ask you for the review?

**Review by dgelessus — COMMENTED** [review]
It's unfortunate that the entire text needs to be packed into a single line like this - both because of JSON and because of Markdown table syntax... This makes it really difficult to review the text in source form and make proper review comments.

Content-wise everything looks good to me. My comments below are only about style and formatting.

A general style thing: with longer descriptions like this, I think it would be better to use complete sentences (capitalized first letter and period at the end). I think "partial sentences" are okay for short descriptions that are only a few words long, but if it's long enough that you already need line breaks to separate the sentences, you might as well use proper capitalization/punctuation.

In the table you have a few places that use `<i>text</i>` tags - any particular reason for not using Markdown's usual italics syntax (`*text*`) instead?

One possible workaround for Markdown not supporting line breaks in tables would be to write the tables in HTML syntax instead, which allows for more control over the formatting (including adding line breaks in table cells, using `<br/>` or `<p>...</p>`. The main disadvantage of this is that Markdown [does not allow using Markdown syntax nested in block-level HTML tags](https://daringfireball.net/projects/markdown/syntax#html). That is, inside the HTML table you would have to use `<a>` tags instead of Markdown links, `<code>` instead of backticks, etc., which is quite inconvenient.

**Comment by Mingun** [comment]
Maybe first convert schema itself to YAML syntax? Then diffs would more friendly

**Comment by dgelessus** [comment]
That would definitely help with making the descriptions more readable. YAML can be converted to JSON quite easily if you stick to a certain subset of features, mainly using only string keys in objects/dictionaries and not using any fancy YAML features like references, tags, multiple documents, etc.

The main problem with using YAML for the schema is that then you always need to perform an extra conversion/compilation step before you can use the schema, which makes using and developing the schema less straightforward. With the current JSON-based schema, you can download the schema JSON or clone the repo and directly load the schema with your editor or other tool that uses it. Or if your editor/tool supports it, you can load the schema directly from [this URL](https://raw.githubusercontent.com/kaitai-io/ksy_schema/master/ksy_schema.json) without even having to clone/download it.

If we switched to using YAML for the schema, this wouldn't be possible anymore - every user would have to download the YAML version of the schema and locally convert it to JSON before being able to use it. Developing the schema would be less convenient as well, because you have to reconvert the schema every time you want to test a change, unlike with the current JSON-based schema, where changes take effect as soon as you save the schema file.

This could be partially solved by always keeping a JSON-converted version alongside the YAML-based schema, but then you need to make sure that every commit updates both the YAML and JSON versions in exactly the same way. Or you could set up a GitHub Actions workflow that does the YAML-to-JSON conversion for every commit and uploads the JSON version onto kaitai.io (or somewhere else), but then it's less obvious where users can find a usable (i. e. JSON) version of the schema, and if you want anything other than the latest version, you have to convert it yourself.

TLDR: It would be great if we could use YAML here, but it might be too impractical to be worth it.

**Comment by Mingun** [comment]
> If we switched to using YAML for the schema, this wouldn't be possible anymore - every user would have to download the YAML version of the schema and locally convert it to JSON before being able to use it.

But why? [There at least several YAML validation libraries](https://github.com/search?q=yaml+schema+validator)

**Comment by dgelessus** [comment]
> > If we switched to using YAML for the schema, this wouldn't be possible anymore - every user would have to download the YAML version of the schema and locally convert it to JSON before being able to use it.
> 
> But why? [There at least several YAML validation libraries](https://github.com/search?q=yaml+schema+validator)

That list is misleading - many of those YAML schema validators don't use JSON Schema, but either have their own YAML-based schema syntax or a DSL in the library's language. Searching for "[json schema yaml](https://github.com/search?q=json+schema+yaml&type=Repositories)" isn't very helpful either, it gives very similar results.

I could only find one library that supports JSON schemas in YAML format: https://github.com/networknt/json-schema-validator. RedHat's YAML extension for VS Code does not support YAML-based JSON schemas (see redhat-developer/vscode-yaml#182). I'm not sure about IntelliJ (can't check at the moment, because I have a beta version installed where the JSON schema support seems to be broken). [Docson](https://github.com/lbovet/docson) (which we now use to generate a human-readable HTML version of the KSY schema) doesn't support YAML-based JSON schemas either.

**Review by GreyCat — APPROVED** [review]
